runtime/bindings/python/src/pyopenvino/core/executable_network.cpp

```diff
@@ -37,9 +37,81 @@ void regclass_ExecutableNetwork(py::module m) {
             Common::set_request_tensors(request, inputs);
             request.infer();
 
-            Containers::InferVec results;
+            Containers::InferMap results;
             for (const auto& out : self.outputs()) {
-                results.push_back(std::make_pair(out, request.get_tensor(out)));
+                ov::runtime::Tensor t{request.get_tensor(out)};
+                switch(t.get_element_type()) {
```

Comment:
Firstly, you may create separate function  in the `common.cpp` file and move this switch there.
Secondly, try to put values into `py::dict` like 
`
py::dict results;
results[out]=array
`

Reply:
Updated:)